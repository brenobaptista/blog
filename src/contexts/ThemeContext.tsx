import { createContext } from 'react'
import { DefaultTheme } from 'styled-components'

import dracula from '@/styles/themes/dracula'

interface IThemeContext {
  theme: DefaultTheme
  toggleTheme: () => void
}

const ThemeContext = createContext<IThemeContext>({
  theme: dracula,
  toggleTheme: () => null
})

export default ThemeContext
