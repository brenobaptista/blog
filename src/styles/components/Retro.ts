import styled from 'styled-components'

export const Marquee = styled.div`
  overflow: hidden;
  position: relative;
  height: 24px;
  margin: 32px 0;

  @media screen and (max-width: 768px) {
    display: none;
  }

  p {
    position: absolute;
    width: 100%;
    height: 100%;
    margin: 0;
    line-height: 24px;
    background-image: linear-gradient(
      to right,
      red,
      orange,
      yellow,
      green,
      blue,
      indigo,
      violet
    );
    background-clip: text;
    -webkit-background-clip: text;
    color: transparent;
    transform: translateX(100%);
    animation: scroll-left 15s linear infinite;
  }

  @keyframes scroll-left {
    0% {
      transform: translateX(100%);
    }
    100% {
      transform: translateX(-100%);
    }
  }
`

export const Construction = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  margin-bottom: 32px;
`

export const Song = styled.figure`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 8px;
`
