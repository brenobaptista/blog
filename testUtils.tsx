import { render } from '@testing-library/react'
import { RenderResult } from '@testing-library/react/types'
import { ThemeProvider } from 'styled-components'

import light from './src/styles/themes/light'

const Providers = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
)

const customRender = (ui: React.ReactElement, options = {}): RenderResult =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
