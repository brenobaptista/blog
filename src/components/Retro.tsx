import { useContext } from 'react'

import ThemeContext from '../contexts/ThemeContext'
import { Marquee, Construction, Badges } from '../styles/components/Retro'

const Retro = (): JSX.Element => {
  const { theme } = useContext(ThemeContext)

  return (
    <>
      {theme.mode === 'retro' && (
        <>
          <Marquee>
            <p>WARNING! This site is currently under heavy construction.</p>
          </Marquee>
          <Construction>
            <img
              src='/static/images/retro/man-jackhammer.gif'
              alt='Man with jackhammer'
              width='100'
              height='113'
            />
            <img
              src='/static/images/retro/construction.gif'
              alt='Construction'
              width='90'
              height='95'
            />
            <img
              src='/static/images/retro/construction2.gif'
              alt='Construction2'
              width='100'
              height='113'
            />
            <img
              src='/static/images/retro/worldwideweb.gif'
              alt='World Wide Web'
              width='91'
              height='100'
            />
          </Construction>
          <Badges>
            <img
              src='/static/images/retro/badges/bookmark.gif'
              alt='Bookmark this page'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/guestbook.gif'
              alt='Sign my guestbook'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/best-viewed.gif'
              alt='Best viewed with anything'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/free-speech.gif'
              alt='Free speech forever'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/nuke-ie.gif'
              alt='Nuke IE'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/anarchy.gif'
              alt='Anarchy now'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/viewed-eyes.gif'
              alt='Best viewed with eyes'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/penguin.gif'
              alt='Tux - Linux penguin'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/geocities.gif'
              alt='Geocities'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/privacy.gif'
              alt='Internet privacy'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/wiby.gif'
              alt='Wiby search engine'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/y2k.gif'
              alt='I survived Y2K'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/email.gif'
              alt='Email me'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/monitor.gif'
              alt='Best viewed with a monitor'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/2001.gif'
              alt='I wish it was 2001'
              width='88'
              height='31'
            />
            <img
              src='/static/images/retro/badges/nocodeandrun.gif'
              alt='No code and run'
              width='88'
              height='31'
            />
          </Badges>
        </>
      )}
    </>
  )
}

export default Retro
