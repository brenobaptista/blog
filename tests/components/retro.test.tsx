import { render } from '../../testUtils'
import Retro from '../../src/components/Retro'
import ThemeContext from '../../src/contexts/ThemeContext'

const renderWithContext = (ui, value) =>
  render(<ThemeContext.Provider value={value}>{ui}</ThemeContext.Provider>)

test('matches snapshot', () => {
  const theme = { mode: 'retro' }
  const { asFragment } = renderWithContext(<Retro />, { theme })

  expect(asFragment()).toMatchSnapshot()
})
