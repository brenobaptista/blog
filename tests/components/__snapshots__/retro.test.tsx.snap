// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  overflow: hidden;
  position: relative;
  height: 24px;
  margin: 32px 0;
}

.c0 p {
  position: absolute;
  width: 100%;
  height: 100%;
  margin: 0;
  line-height: 24px;
  background-image: linear-gradient( to right, red, orange, yellow, green, blue, indigo, violet );
  background-clip: text;
  -webkit-background-clip: text;
  color: transparent;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
  -webkit-animation: scroll-left 15s linear infinite;
  animation: scroll-left 15s linear infinite;
}

@media screen and (max-width:768px) {
  .c0 {
    display: none;
  }
}

<div
    class="c0"
  >
    <p>
      WARNING! This site is currently under heavy construction.
    </p>
  </div>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  margin-bottom: 32px;
}

@media screen and (max-width:768px) {

}

<div
    class="c0"
  >
    <img
      alt="Man with jackhammer"
      height="100"
      src="/images/retro/jackhammer.gif"
    />
    <img
      alt="Dancing hamster"
      height="100"
      src="/images/retro/hamster.gif"
    />
    <img
      alt="Spinning money"
      height="100"
      src="/images/retro/money.gif"
    />
    <img
      alt="World Wide Web"
      height="100"
      src="/images/retro/worldwideweb.gif"
    />
  </div>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

@media screen and (max-width:768px) {

}

<figure
    class="c0"
  >
    <figcaption>
      Unleash the champion within:
    </figcaption>
    <audio
      controls=""
    >
      <source
        src="/audio/champions.ogg"
        type="audio/ogg"
      />
      <source
        src="/audio/champions.mp3"
        type="audio/mpeg"
      />
      Your browser does not support the audio element.
    </audio>
  </figure>
</DocumentFragment>
`;
